Clean up and simplify pointer documentation

Troy-Ferrell opened the pull request:
## Overview
This change cleans and updates much of the Pointer documentation. Originally there were three different pages for pointers and each had overlapping but slightly different content. This organizes the content into Pointer Architecture and Pointers in Practice. 

- Added links to pointer classes on docs via xref
- Add code snippets for how to do key things
- Fixed some image resizing
- Fixed some formating, line spacing, etc.
- Added additional information on how to configure pointers via profiles
- Added summary tag and tooltips to some code
- Deprecates README_Pointer.md, removal from TOC and updates on getting started/readme pages

## Changes
- Fixes: #4953 

## Verification

> As a reviewer, it is possible to check out this change locally by using the following
> commands (substituting {PR_ID} with the ID of this pull request):
>
> git fetch origin pull/{PR_ID}/head:name_of_local_branch
>
> git checkout name_of_local_branch

cre8ivepark commented:
I have updated some old images and added an overview image for the pointers. Please let me know if you see anything need to be updated.
![MRTK_Pointers](https://user-images.githubusercontent.com/13754172/62981225-92cf6c00-bddd-11e9-82ab-badcb2a8e3a1.png)

Troy-Ferrell commented:
TouchPoint

> I have updated some old images and added an overview image for the pointers. Please let me know if you see anything need to be updated.

Awesome! Few thoughts.

1) Pointer mediator is actually a bit tricky. I was reading into the code more and the PointerMediator is created per input source. So it's weird to show the pointer mediator connecting all the pointers when you wouldn't have pokepointer with parabolic pointer on the same input source.

2) TouchPointer is for touchscreens yes? The hand image suggests articulated hand. I'd rather show like a phone icon or surface with a hand etc

cre8ivepark commented:
Updated version:
![MRTK_Pointers](https://user-images.githubusercontent.com/13754172/62987296-eb116880-bdf3-11e9-9e21-a958c926f761.png)

wiwei commented on Documentation/Input/Pointers.md:
```diff
@@ -1,39 +1,239 @@
+
+![Pointer Profile Example](../Images/Input/Pointers/PointerProfile.png)
+
+The default MRTK Pointer Profile configuration includes the following pointer classes and associated prefabs out-of-box. The list of pointers available to the system at runtime is defined under *Pointer Options* in the Pointer profile. Developers can utilize this list to reconfigure existing Pointers, add new Pointers, or delete one.
```
>following [](start = 60, length = 9)

This instance of the word following feels like it's a bit out of place, because the text that follows this doesn't contain a list of pointer classes of associated prefabs - maybe this should be "preceeding" because it's referring to the content in the picture?

Troy-Ferrell replied:
Move image back up to make sense

wiwei commented on Documentation/Input/Pointers.md:
```diff
@@ -1,39 +1,239 @@
+
+##### [`LinePointer`](xref:Microsoft.MixedReality.Toolkit.Input.MousePointer)
+
+ A base pointer class, draws lines from the source of the input (i.e. the controller) in the pointer direction. This is also used by other things like the Shell Hand Ray Pointer and the teleport pointers (which also draw lines to indicate where teleportation will end up at).
```
It might be worth adding a note that this pointer itself isn't directly instantiated, and is only used through subclassing of the shell hand ray pointer and teleport pointers.

The reason I mention this is because the next line contains additional content (i.e. rotation stuff) which makes me think that actually this is more useful itself than it is (i.e. DefaultControllerPointer isn't mentioned here, which is actually the main one that is actually referenced in the profile.)

Troy-Ferrell replied:
Gah, and it's totally wrong reference

Troy-Ferrell replied:
Also DefaultControllerPointer is only the prefab which is mentioned earlier on and is mentioned with ShellHandRayPointer

wiwei commented on Documentation/Input/Pointers.md:
```diff
@@ -1,39 +1,239 @@
+
+#### Register globally
+
+blah need to add code example for CoreServices.InputSystem.Register etc
```
It feels like this section should probably just point to a general "how to register for input events" instead of having to necessarily copy paste and have another define of that... so maybe consider having a reference to somewhere else (which I don't think exists FWIW).

Troy-Ferrell replied:
Ah I see what you are talking about now

wiwei commented on Documentation/Input/Pointers.md:
```diff
@@ -1,39 +1,239 @@
+    private void OnDisable()
+    {
+        CoreServices.InputSystem?.FocusProvider?.UnsubscribeFromPrimaryPointerChanged(OnPrimaryPointerChanged);
+        OnPrimaryPointerChanged(null, null);
```
>OnPrimaryPointerChanged(null, null); [](start = 8, length = 36)

Nit: Add a comment stating that this is designed to flush out the current primary pointer

wiwei commented:
/azp run mrtk_docs

Troy-Ferrell commented:
> I found that some of the new images has .PNG instead of .png which will be broken on HTML version. I tried to change the file extensions to lower case but git did not detect them as changed file. Could you please check? Or we might have to update the .md file to use proper upper case.
> 
> Broken image example:
> https://github.com/microsoft/MixedRealityToolkit-Unity/blob/cd86fbf5d08a179982383b128e67e3a394f79c26/Documentation/Input/Pointers.md

Got the fix in!

Troy-Ferrell commented:
/azp run

thalbern commented on Documentation/Input/Pointers.md:
```diff
@@ -1,39 +1,235 @@
+The MousePointer powers a screen to world raycast for far interactions, but for mouse instead of touch.
+
+<img src="../../Documentation/Images/Pointers/MRTK_MousePointer.png" width="400">
+
```
` [!NOTE] Mouse support is not available by default in MRTK and can be enabled by adding a new *Input Data Provider* of type @Microsoft.MixedReality.Toolkit.Input.UnityInput.MouseDeviceManager to the MRTK input profile.`

Troy-Ferrell commented on Documentation/README_Pointers.md:
```diff
@@ -1,76 +1,5 @@
-# Pointers #
+# Pointer #
```
Because there may still be links out there referencing it. It's a pretty old file. Thus it's basically in a deprecated state and taken off the TOC until the next iteration. Just like all of the [Obsolete] tags

Troy-Ferrell commented:
/azp run

thalbern commented:
doc build failed : (Documentation/Input/Pointers.md)Text in the first line of Note/Section/Video is not valid. Will be rendererd to <blockquote>

Troy-Ferrell commented:
/azp run

Troy-Ferrell commented:
/azp run
